driver-core/src/main/com/mongodb/internal/Locks.java

```diff
@@ -52,6 +52,7 @@ public static void lockInterruptibly(final Lock lock) throws MongoInterruptedExc
         try {
             lock.lockInterruptibly();
         } catch (InterruptedException e) {
+            lock.unlock();
```

Comment:
If `InterruptedException` is thrown from `lockInterruptibly`, how is it not already unlocked (unless it's been locked by a different thread, in which case that thread should be responsible for unlocking)?

Reply:
This `lock.unlock()` will cause `IllegalMonitorStateException` if `ReentrantLock`/`ReentrantReadWriteLock` is used. If `InterruptedException` is thrown from `Lock.lockInterruptibly`, then the lock hasn't been acquired by the current thread.